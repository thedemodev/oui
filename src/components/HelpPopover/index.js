import PropTypes from 'prop-types';
import React from 'react';
import OverlayWrapper from '../OverlayWrapper';
import Popover from '../Popover';
import Icon from 'react-oui-icons';

const HelpPopover = ({
  behavior,
  children,
  horizontalAttachment,
  iconSize,
  isConstrainedToScreen,
  popoverTitle,
  verticalAttachment,
  testSection,
}) => (
  <OverlayWrapper
    behavior={ behavior }
    horizontalAttachment={ horizontalAttachment }
    isConstrainedToScreen={ isConstrainedToScreen }
    overlay={
      <Popover title={ popoverTitle }>
        { children }
      </Popover>
    }
    testSection={ testSection }
    verticalAttachment={ verticalAttachment }>
    <span
      className="cursor--pointer push-half--left"
      style={{'marginBottom': '1px'}}
      data-test-section="help-popover-icon">
      <Icon
        name="help"
        size={ iconSize }
      />
    </span>
  </OverlayWrapper>
);

HelpPopover.propTypes = {
  /** Event to listen to and open the overlay */
  behavior: PropTypes.oneOf(['click', 'hover']),
  /** Description of thing that the person hovered for. */
  children: PropTypes.node.isRequired,
  /** Side of the `overlay` that should attach to the `children` */
  horizontalAttachment: PropTypes.oneOf(['left', 'center', 'right']),
  /** Size of icon */
  iconSize: PropTypes.oneOf(['small', 'medium', 'large']),
  /** Attach `overlay` to an edge of the screen if it is going to move off */
  isConstrainedToScreen: PropTypes.bool,
  /** Title of thing that is being described. */
  popoverTitle: PropTypes.string,
  /** Test section */
  testSection: PropTypes.string,
  /** Vertical edge of the `overlay` that should touch the `children` */
  verticalAttachment: PropTypes.oneOf(['top', 'middle', 'bottom']),
};

HelpPopover.defaultProps = {
  behavior: 'click',
  iconSize: 'medium',
  isConstrainedToScreen: true,
  testSection: 'help-popover',
};

export default HelpPopover;
