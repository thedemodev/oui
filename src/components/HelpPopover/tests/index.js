import React from 'react';
import HelpPopover from '../index.js';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe.only('components/HelpPopover', () => {
  it('renders correctly', () => {
    const component = shallow(<HelpPopover popoverTitle="I am a popover title!">I am a popover description.</HelpPopover>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render text passed in as children and with default props', function() {
    const message = 'Hello! This is a short popover with no title, just the defaults.';
    const component = mount(
      <HelpPopover>{ message }</HelpPopover>
    );
    const overlayWrapper = component.find('OverlayWrapper');
    expect(overlayWrapper.length).toBe(1);
    expect(overlayWrapper.props().behavior).toBe('click');
    expect(overlayWrapper.props().isConstrainedToScreen).toBe(true);
    expect(overlayWrapper.props().testSection).toBe('help-popover');

    const popover = component.find('Popover');
    expect(popover.length).toBe(1);
    expect(popover.props().title).toBe(undefined);
    expect(popover.props().children).toBe(message);
  });

  it('should render custom props properly', function() {
    const child = (
      <p>
        I am some random popover content. ðŸ¤“
      </p>
    );

    const component = mount(
      <HelpPopover
        behavior="hover"
        testSection="custom-test-section"
        popoverTitle="I am a title">
        { child }
      </HelpPopover>
    );
    const overlayWrapper = component.find('OverlayWrapper');
    expect(overlayWrapper.length).toBe(1);
    expect(overlayWrapper.props().behavior).toBe('hover');

    const popover = component.find('Popover');
    expect(popover.length).toBe(1);
    expect(popover.props().title).toBe('I am a title');
    expect(popover.props().children).toBe(child);
  });
});
