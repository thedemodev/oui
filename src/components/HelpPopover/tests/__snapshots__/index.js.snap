// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HelpPopover renders correctly 1`] = `
<OverlayWrapper
  attachmentConstraint="together"
  behavior="click"
  horizontalAttachment="center"
  isConstrainedToScreen={true}
  onHide={[Function]}
  overlay={
    <Popover
      padding="default"
      title="I am a popover title!"
    >
      I am a popover description.
    </Popover>
  }
  shouldHideOnClick={true}
  testSection="help-popover"
  verticalAttachment="top"
>
  <span
    className="cursor--pointer push-half--left"
    data-test-section="help-popover-icon"
    style={
      Object {
        "marginBottom": "1px",
      }
    }
  >
    <Icon
      name="help"
      size="medium"
    />
  </span>
</OverlayWrapper>
`;
